src/js/helpers/ProjectOverwriteHelpers.js

```diff
@@ -87,33 +94,107 @@ export const getProjectName = (projectPath) => {
+
+export function validateSelectionsForTool(projectSaveLocation, chapter, verse, bookId, targetVerse, userName, toolName) {
+  return (dispatch, getState) => {
+    const contextId = {
+      reference: {
+        bookId,
+        chapter: parseInt(chapter),
+        verse: parseInt(verse)
+      }
+    };
+    const groupsData = loadProjectGroupData(toolName, projectSaveLocation);
+    const groupsDataForVerse = getGroupDataForVerse(groupsData, contextId, name);
+    let filtered = null;
+    let selectionsChanged = false;
+    for (let groupItemKey of Object.keys(groupsDataForVerse)) {
+      const groupItem = groupsDataForVerse[groupItemKey];
+      for (let checkingOccurrence of groupItem) {
+        const selections = checkingOccurrence.selections;
+        if (!sameContext(contextId, checkingOccurrence.contextId)) {
+          if (selections && selections.length) {
+            if (!filtered) {  // for performance, we filter the verse only once and only if there is a selection
+              filtered = usfm.removeMarker(targetVerse); // remove USFM markers
+            }
+            const validSelections = checkSelectionOccurrences(filtered, selections);
+            if (selections.length !== validSelections.length) {
+              const selectionsObject = getSelectionsFromChapterAndVerseCombo(
+                bookId,
+                chapter,
+                verse,
+                projectSaveLocation,
+                checkingOccurrence.contextId.quote
+              );
+              //If selections are changed, they need to be clearded
+              selectionsChanged = true;
+              const invalidatedCheckPath = path.join(projectSaveLocation, '.apps', 'translationCore', 'checkData', 'invalidated', bookId, chapter.toString(), verse.toString());
+              const invalidatedPayload = {
+                ...selectionsObject,
+                invalidated: true,
+                selections: [],
+                userName
+              };
+              writeCheckData(invalidatedPayload, invalidatedCheckPath);
+
+              const selectionsCheckPath = path.join(projectSaveLocation, '.apps', 'translationCore', 'checkData', 'selections', bookId, chapter.toString(), verse.toString());
+              const selectionsPayload = {
+                ...selectionsObject,
+                selections: [],
+                userName
+              };
+              writeCheckData(selectionsPayload, selectionsCheckPath);
+            }
+          }
         }
-      });
-    } catch (error) {
-      console.log(error);
+      }
     }
-  });
-};
+    if (selectionsChanged) {
+      const translate = getTranslate(getState());
+      dispatch(AlertActions.openIgnorableAlert('selections_reset', translate('tools.selections_invalidated')));
+    }
+  };
+}
 
-export const createExternalVerseEdit = (projectPath, verseBefore, verseAfter, bookId, chapter, verse, userName, modifiedTimestamp) => {
+function writeCheckData(payload = {}, checkPath) {
```

Comment:
If we have an invalidated reducer you should not need this function since it will be autosaved? Am my missing something?

Reply:
You are correct, it would ideal to call the action as the usual workflow such as we do here https://github.com/unfoldingWord-dev/translationCore/blob/develop/src/js/actions/SelectionsActions.js#L228

But the problem is that this use case is before a project or tool is even selected, the autosave assumes you already have that data loaded in the state.